posts/2022-02-17-concurrency-3.0-22003-beta.adoc

```diff
@@ -0,0 +1,197 @@
+
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import javax.sql.DataSource;
+
+@Singleton
+public class AccountsBean {
+
+    @Asynchronous(executor = "java:module/concurrent/MyExecutor")
+    CompletableFuture<List<Account>> findOverdue(int minDaysLate) {
+        List<Account> overdueAccounts = new ArrayList<Account>();
+        try {
+            // Resource reference lookup requires context of the application component,
+            DataSource db = InitialContext.doLookup("java:comp/env/jdbc/account-db-ref");
+            ...
+            return Asynchronous.Result.complete(overdueAccounts);
+        } catch (NamingException | SQLException x) {
+            throw new CompletionException(x);
+        }
+    }
+
+    ...
+----
+
+When the application invokes the asynchronous method, the container intercepts the invocation and hands back a completion stage representing its eventual completion, which is arranged by the `ManagedExecutorService`.
+
+[source, java]
+----
+    public void doGet(HttpServletRequest req, HttpServletResponse res) {
+        ...
+        accounts.findOverdue(60).thenAccept(this::sendReminder);
+        ...
+    }
+
+    void sendReminder(List<Account> overdue) {
+        try {
+            // Resource reference lookup requires context of the application component,
+            DataSource db = InitialContext.doLookup("java:comp/env/jdbc/customer-db-ref");
+            ...
+        } catch (NamingException | SQLException x) {
+            throw new CompletionException(x);
+        }
+    }
+----
+
+The Concurrency 3.0 specification hasn't been published yet because Jakarta EE 10 is still under development. However, you can preview a draft of it at the link:https://github.com/eclipse-ee4j/concurrency-api/blob/2abe74ce30589ef53e248009fcb446491fd42a27/specification/src/main/asciidoc/jakarta-concurrency.adoc[concurrency-api GitHub page].
+
+Similarly, the API jar will not be available on Maven until the specification is released, but in the meantime, you can check out the link:http://public.dhe.ibm.com/ibmdl/export/pub/software/olrepo/io/openliberty/jakarta/enterprise/concurrent/jakarta.enterprise.concurrent-api/3.0.0.20220120/[draft copy.]
+
+[source, xml]
+----
+    <repositories>
+        <repository>
+            <id>ibmdhe</id>
+            <name>IBM_DHE File Server</name>
+            <url>https://public.dhe.ibm.com/ibmdl/export/pub/software/olrepo</url>
+        </repository>
+    </repositories>
+    <dependencies>
+        <dependency>
+            <groupId>io.openliberty.jakarta.enterprise.concurrent</groupId>
+            <artifactId>jakarta.enterprise.concurrent-api</artifactId>
+            <version>3.0.0.20220120</version>
+        </dependency>
+    <dependencies>
+----
+
+=== Try it now 
+
+To try out these features, just update your build tools to pull the Open Liberty All Beta Features package instead of the main release. The beta works with Java SE 15, Java SE 11, or Java SE 8.
```

Comment:
```suggestion
To try out these features, just update your build tools to pull the Open Liberty All Beta Features package instead of the main release. The beta works with Java SE 17, Java SE 11, or Java SE 8.
```